### PR TITLE
Fix for chart names in dropdown select box.

### PR DESCRIPTION
**Pull request by smartguest**
This moves the check of alternative names from selectbox to chartview as its more appropriate there. Any attempts to change the name of the options resulted in a failure to load the chart itself so a separate array is passed into selectbox with the user-friendly names in order not to cause conflict with the existing code. Also the array map has been replaced with a string to string map.

### CODE REVIEW

File: src/sql/workbench/parts/charts/browser/chartView.ts [hunk 1]
```diff
@@ -32,6 +32,21 @@ declare class Proxy {
 
 const insightRegistry = Registry.as<IInsightRegistry>(Extensions.InsightContribution);
 
+//Map used to store names and alternative names for chart types.
+//This is mainly used for comparison when options are parsed into the constructor.
+const altNameHash: { [oldName: string]: string } = {
+	'horizontalBar': 'Horziontal Bar',
```
Comment (Charles-Gagnon) [inline src/sql/workbench/parts/charts/browser/chartView.ts]: If these are being displayed to the user they should be localized

File: src/sql/workbench/parts/charts/browser/chartView.ts [hunk 2]
```diff
@@ -127,6 +143,13 @@ export class ChartView extends Disposable implements IPanelView {
 
 	}
 
+	// method that is used to generate array of user-friendly ones for display.
+	private changeToAltNames(option: string[]): string[] {
+		let newArray: string[] = option.slice();
+		newArray.forEach(function (value, index) { if (altNameHash[value] !== undefined) { newArray[index] = altNameHash[value]; } });
```
Comment (Charles-Gagnon) [inline src/sql/workbench/parts/charts/browser/chartView.ts]: map seems like a better choice here 

`return option.map(o => altNameHash[o] || o);`

File: src/sql/workbench/parts/charts/browser/chartView.ts [hunk 3]
```diff
@@ -127,6 +143,13 @@ export class ChartView extends Disposable implements IPanelView {
 
 	}
 
+	// method that is used to generate array of user-friendly ones for display.
```
Comment (Charles-Gagnon) [inline src/sql/workbench/parts/charts/browser/chartView.ts]: Make sure you're using actual JSDoc comments - https://jsdoc.app/about-getting-started.html

If you type /** and hit enter above a method header it'll auto-create the comment for you (with params and whatnot already filled in)

File: src/sql/workbench/parts/charts/browser/chartView.ts [hunk 4]
```diff
@@ -25,13 +25,29 @@ import { CreateInsightAction, CopyAction, SaveImageAction, IChartActionContext }
...
 
 declare class Proxy {
 	constructor(object, handler);
 }
 
 const insightRegistry = Registry.as<IInsightRegistry>(Extensions.InsightContribution);
 
+//Map used to store names and alternative names for chart types.
+//This is mainly used for comparison when options are parsed into the constructor.
+const altNameHash: { [oldName: string]: string } = {
```
Comment (aaomidi) [inline src/sql/workbench/parts/charts/browser/chartView.ts]: Why is oldName in an array here?

**Comment by aaomidi** [comment]
Can you change the PR name too? Right now its just Alex/alexbranch

File: src/sql/workbench/parts/charts/browser/chartView.ts (hunk 4 above)
Comment (aaomidi) [reply src/sql/workbench/parts/charts/browser/chartView.ts]: Nevermind I got confused

**Pull request merged**